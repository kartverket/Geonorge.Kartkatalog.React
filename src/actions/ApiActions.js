export const getApiData = (url) => (dispatch, getState) => {
  const fetchOptions = {
    method: 'GET',
    headers: {
      'Accept': 'application/json',
      'Content-Type': 'application/json'
    }
  };
  return fetch(url, fetchOptions).then(res => res.json()).then(
    apiData => {
      return apiData
    }
  ).catch(error => {
    console.log(error);
  })
}
