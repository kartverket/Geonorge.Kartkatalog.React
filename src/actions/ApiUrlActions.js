export const getKartkatalogApiUrl = () => () => {
    const environmentSlug = process.env.REACT_APP_ENVIRONMENT === 'dev' || process.env.REACT_APP_ENVIRONMENT === 'test' ? process.env.REACT_APP_ENVIRONMENT + '.' : '';
    return `https://kartkatalog.${environmentSlug}geonorge.no/api`;
};

export const getGeonorgeRegisterApiUrl = () => () => {
    const environmentSlug = process.env.REACT_APP_ENVIRONMENT === 'dev' || process.env.REACT_APP_ENVIRONMENT === 'test' ? process.env.REACT_APP_ENVIRONMENT + '.' : '';
    return `https://register.${environmentSlug}geonorge.no/api`;
};

export const getGeonorgeMenuUrl = (selectedLanguage) => () => {
    const environmentSlug = process.env.REACT_APP_ENVIRONMENT === 'dev' || process.env.REACT_APP_ENVIRONMENT === 'test' ? 'test.' : '';
    const selectedLanguageSlug = selectedLanguage === 'en' ? 'en/' : '';
    return `https://www.${environmentSlug}geonorge.no/${selectedLanguageSlug}api/menu?omitLinks=1`;
};

export const getServiceStatusApiUrl = () => () => {
    const environmentSlug = process.env.REACT_APP_ENVIRONMENT === 'dev' || process.env.REACT_APP_ENVIRONMENT === 'test' ? 'testmonitorApi' : 'monitorApi';
    return `https://status.geonorge.no/${environmentSlug}`;
};
