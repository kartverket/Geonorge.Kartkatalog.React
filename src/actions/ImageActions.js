import logoDev from "images/svg/geonorge-navbar-logo_dev.svg";
import logoTest from "images/svg/geonorge-navbar-logo_test.svg";
import logo from "images/svg/geonorge-navbar-logo.svg";

export const getGeonorgeLogo = () => () => {
    const imageSrc = () => {
        switch (process.env.REACT_APP_ENVIRONMENT) {
            case 'dev':
                return logoDev;
            case 'test':
                return logoTest;
            default:
                return logo;
        }
    };
    return imageSrc();
};
