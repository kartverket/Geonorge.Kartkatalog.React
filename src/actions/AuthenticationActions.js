import {UPDATE_BAAT_INFO} from 'actions/types';
import Cookies from 'js-cookie';
import {pushToDataLayer} from 'reducers/TagManagerReducer';

export const updateOidcCookie = () => (dispatch, getState) => {
  const user = getState() && getState().auth && getState().auth.user
    ? getState().auth.user
    : null;
  if (user) {
    const accessToken = user.access_token
      ? user.access_token
      : null;
    const expiresAt = user.expires_at
      ? user.expires_at * 1000
      : null;
    if (accessToken && expiresAt) {
      Cookies.set('oidcAccessToken', accessToken, {expires: new Date(expiresAt)});
    }
  }
};

export const updateBaatInfo = () => (dispatch, getState) => {
  const user = getState() && getState().auth && getState().auth.user
    ? getState().auth.user
    : null;
  const savedBaatInfo = getState() && getState().baatInfo && Object.keys(getState().baatInfo).length
    ? getState().baatInfo
    : null;
  if (user && user.profile && user.profile.preferred_username && !savedBaatInfo) {
    const accessToken = user.access_token
      ? user.access_token
      : null;
    const expiresAt = user.expires_at
      ? user.expires_at * 1000
      : null;
    if (accessToken && expiresAt) {
      const userInfoUrl = `${process.env.REACT_APP_GEOID_BAATAUTHZ_APIURL}info/${user.profile.preferred_username}`;
      fetch(userInfoUrl, {
        method: 'GET',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
          Authorization: 'Bearer ' + accessToken
        }
      }).then((res) => res.json()).then((baatInfo) => {
        Cookies.set('baatInfo', JSON.stringify(baatInfo), {expires: new Date(expiresAt)});
        const baatOrganizationName = baatInfo && baatInfo.baat_organization && baatInfo.baat_organization.name
          ? baatInfo.baat_organization.name
          : null;
        if (baatOrganizationName) {
          dispatch(pushToDataLayer(
            {
              event: 'signIn',
              category: 'baatOrganizationName',
              activity: 'authentication',
              baatOrganizationName: baatOrganizationName
            }
          ));

        }
        dispatch({type: UPDATE_BAAT_INFO, payload: baatInfo});
      });
    }
  }
}
