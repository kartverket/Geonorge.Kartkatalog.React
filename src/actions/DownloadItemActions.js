// Dependencies
import Cookies from "js-cookie";

// Actions
import { FETCH_ITEMS_TO_DOWNLOAD } from "actions/types";

// Reducers
import { pushToDataLayer } from "reducers/TagManagerReducer";

export const fetchItemsToDownload = () => (dispatch) => {
    let itemsToDownload =
        localStorage.orderItems && Array.isArray(JSON.parse(localStorage.orderItems))
            ? JSON.parse(localStorage.orderItems)
            : [];
    dispatch({ type: FETCH_ITEMS_TO_DOWNLOAD, payload: itemsToDownload });
    const event = new Event("downloadItemsChanged");
    document.dispatchEvent(event);
};

export const getDownloadItemMetadata = (itemUuid) => (dispatch) => {
    return localStorage[itemUuid + ".metadata"] ? JSON.parse(localStorage[itemUuid + ".metadata"]) : null;
};

export const removeItemSelectedForDownload = (itemToRemove) => (dispatch) => {
    let selectedItems =
        localStorage.orderItems && Array.isArray(JSON.parse(localStorage.orderItems))
            ? JSON.parse(localStorage.orderItems)
            : [];
    const selectedItemsToKeep = selectedItems.filter((itemToKeep) => itemToKeep !== itemToRemove.uuid);
    Cookies.set("orderItems", selectedItemsToKeep.length, { expires: 7, path: "/", domain: ".geonorge.no" });
    localStorage.orderItems = JSON.stringify(selectedItemsToKeep);

    // TODO midlertidig løsning pga gammel handlekurv...
    localStorage.removeItem(itemToRemove.uuid + ".metadata");
    const tagData = {
        name: itemToRemove.name,
        uuid: itemToRemove.uuid,
        accessIsOpendata: itemToRemove.accessIsOpendata,
        accessIsRestricted: itemToRemove.accessIsRestricted,
        organizationName: itemToRemove.organizationName,
        theme: itemToRemove.theme
    };
    dispatch(
        pushToDataLayer({ event: "updateCart", category: "download", activity: "removeFromCart", metadata: tagData })
    );

    dispatch(fetchItemsToDownload());
};

const addItemToLocalStorage = (itemToAdd => {
  try 
  {

    let selectedItems = localStorage.orderItems && Array.isArray(JSON.parse(localStorage.orderItems))
      ? JSON.parse(localStorage.orderItems)
      : [];
    selectedItems.push(itemToAdd.uuid);
    Cookies.set('orderItems', selectedItems.length, { expires: 7, path: '/', domain: '.geonorge.no' });
    localStorage.orderItems = JSON.stringify(selectedItems);
    // TODO midlertidig løsning pga gammel handlekurv...
    localStorage.setItem(itemToAdd.uuid + ".metadata", JSON.stringify(itemToAdd))
  }
  catch (e) {
      console.log("Local Storage is full");
      itemToAdd.areas = {};
      localStorage.setItem(itemToAdd.uuid + ".metadata", JSON.stringify(itemToAdd))
  }
});

export const addItemSelectedForDownload = (itemToAdd) => (dispatch, getState) => {
    if (itemToAdd.accessIsOpendata) {
        addItemToLocalStorage(itemToAdd);
        const tagData = {
            name: itemToAdd.name,
            uuid: itemToAdd.uuid,
            accessIsOpendata: itemToAdd.accessIsOpendata,
            accessIsRestricted: itemToAdd.accessIsRestricted,
            organizationName: itemToAdd.organizationName,
            theme: itemToAdd.theme
        };
        dispatch(
            pushToDataLayer({ event: "updateCart", category: "download", activity: "addToCart", metadata: tagData })
        );
        dispatch(fetchItemsToDownload());
    } else if (itemToAdd.accessIsRestricted) {
        const baatInfo = getState() && getState().baatInfo ? getState().baatInfo : null;
        if (baatInfo) {
            const roles = baatInfo.baat_services ? baatInfo.baat_services : null;
            const constraintRequiredRole = itemToAdd.capabilities.accessConstraintRequiredRole;
            let addDatasetIsAllowed = true;
            if (constraintRequiredRole === undefined) {
                addDatasetIsAllowed = true;
            } else {
                const requiredRoles = constraintRequiredRole.split(",").map((item) => item.trim());
                for (let requiredRole of requiredRoles) {
                    addDatasetIsAllowed = requiredRole
                        ? roles &&
                          roles.length &&
                          roles.find((role) => {
                              return role === requiredRole;
                          }) !== undefined
                        : true;
                    if (addDatasetIsAllowed) break;
                }
                let isAdmin =
                    roles &&
                    roles.length &&
                    roles.find((role) => {
                        return role === "nd.metadata_admin";
                    }) !== undefined;
                if (isAdmin) addDatasetIsAllowed = true;
            }
            if (addDatasetIsAllowed) {
                addItemToLocalStorage(itemToAdd);
                dispatch(fetchItemsToDownload());
            } else {
                alert("Du har ikke tilgang til å legge datasett til nedlasting");
            }
        }
    }
};

export const autoAddItemFromLocalStorage = () => (dispatch, getState) => {
    const hasItemToAdd =
        localStorage.autoAddDownloadItemOnLoad &&
        localStorage.autoAddDownloadItemOnLoad.length &&
        localStorage.autoAddDownloadItemOnLoad !== "undefined" &&
        localStorage.autoAddDownloadItemOnLoad !== "null";
    const itemToAdd = hasItemToAdd ? JSON.parse(localStorage.autoAddDownloadItemOnLoad) : null;
    const isLoggedIn = getState().oidc && getState().oidc.user;
    const hasBaatInfo = getState().baatInfo && getState().baatInfo.user;
    if (itemToAdd && isLoggedIn && hasBaatInfo) {
        dispatch(addItemSelectedForDownload(itemToAdd));
        localStorage.removeItem("autoAddDownloadItemOnLoad");
    }
};
