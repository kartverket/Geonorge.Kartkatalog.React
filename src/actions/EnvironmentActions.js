import { GET_ENVIRONMENT } from 'actions/types';

export const getEnvironment = () => dispatch => {
    const environment =  {
      buildNumber: process.env.REACT_APP_BUILD_NUMBER || null,
      majorMinorVersionMaster: process.env.REACT_APP_MAJOR_MINOR_VERSION_MASTER || null,
      majorMinorVersionDevelopment: process.env.REACT_APP_MAJOR_MINOR_VERSION_DEVELOPMENT || null,
      environment: process.env.REACT_APP_ENVIRONMENT || null
    };
    dispatch({
        type: GET_ENVIRONMENT,
        payload: environment
    });
}
