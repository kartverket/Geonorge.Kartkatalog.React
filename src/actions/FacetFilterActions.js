import { ADD_SELECTEDFACET, REMOVE_SELECTEDFACET } from '../actions/types';

export const addSelectedFacet = (facet, facetField) => dispatch => {
	dispatch({
		type: ADD_SELECTEDFACET,
		payload: facet,
		facetField: facetField
	})
}

export const removeSelectedFacet = (facet, facetField) => dispatch => {
	dispatch({
		type: REMOVE_SELECTEDFACET,
		payload: facet,
		facetField: facetField
	})
}
