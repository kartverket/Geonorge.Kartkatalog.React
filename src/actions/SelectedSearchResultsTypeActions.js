import { UPDATE_SELECTED_SEARCHRESULTS_TYPE } from '../actions/types';

export const updateSelectedSearchResultsType = (searchResultsType) => dispatch => {
    dispatch({
        type: UPDATE_SELECTED_SEARCHRESULTS_TYPE,
        payload: searchResultsType,
    })
}