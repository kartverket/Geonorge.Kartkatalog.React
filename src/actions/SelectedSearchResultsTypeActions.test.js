import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk'

// Actions to be tested
import * as selectedSearchResultsTypeActions from 'actions/SelectedSearchResultsTypeActions';

const middlewares = [thunk] // add your middlewares like `redux-thunk`
const mockStore = configureStore(middlewares);
const store = mockStore();

describe('update_selected_search_results_type_actions', () => {
    beforeEach(() => { // Runs before each test in the suite
        store.clearActions();
    });

    describe('updateSelectedSearchResultsType', () => {
        test('Dispatches the correct action and payload', () => {
            const searchResultsType = 'software';
            store.dispatch(selectedSearchResultsTypeActions.updateSelectedSearchResultsType(searchResultsType));
            expect(store.getActions()).toMatchSnapshot();
        });
    });
});
