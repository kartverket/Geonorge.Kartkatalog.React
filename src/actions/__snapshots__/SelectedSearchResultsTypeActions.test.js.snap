// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`update_selected_search_results_type_actions updateSelectedSearchResultsType Dispatches the correct action and payload 1`] = `
Array [
  Object {
    "payload": "software",
    "type": "UPDATE_SELECTED_SEARCHRESULTS_TYPE",
  },
]
`;
